zerver/views/realm.py

```diff
@@ -143,6 +143,9 @@ def update_realm(
     ),
     org_type: Optional[int] = REQ(json_validator=check_int_in(ORG_TYPE_IDS), default=None),
     enable_spectator_access: Optional[bool] = REQ(json_validator=check_bool, default=None),
+    list_on_public_communities_directory: Optional[bool] = REQ(
+        json_validator=check_bool, default=None
+    ),
```

Comment:
The API docs say this can only be enabled in certain situations; but I think the correct story is that it only does something useful in those situations, and the API docs should instead describe (1) that it only does something in those cases and (2) that settings UI should only offer it in appropriate situations.

Reply:
Agreed. I've updated the descriptions (and screenshots) of the API documentation for these changes.